test/unit/conftest.py

```diff
@@ -327,7 +335,83 @@ def _mock(side_effects, existing_device=None, device=ios_device):
 def netmiko_device():
     # with mock.patch.object(BaseNetmikoDevice, "confirm_is_active") as mock_confirm:
     #     mock_confirm.return_value = True
-    with mock.patch("netmiko.ConnectHandler") as ch:
-        device = BaseNetmikoDevice("host", "user", "password", "cisco_ios_ssh")
-        device.native = ch
+    with mock.patch("netmiko.base_connection.BaseConnection", autospec=True) as base_connection:
+        with mock.patch.object(BaseNetmikoDevice, "__init__", return_value=None):
+            device = BaseNetmikoDevice()
+            device.native = base_connection
+        return device
+
+
+@pytest.fixture
+def netmiko_command(netmiko_device, netmiko_mock_path):
+    def _mock(netmiko_method, side_effects, existing_device=None, device=netmiko_device):
+        if existing_device is not None:
+            device = existing_device
+        side_effects = get_side_effects(netmiko_mock_path, side_effects)
+        native_method = getattr(device.native, netmiko_method)
+        native_method.side_effect = side_effects
+        return device
+
+    return _mock
+
+
+@pytest.fixture
+def netmiko_mock_path(mock_path):
+    return f"{mock_path}/netmiko"
+
+
+@pytest.fixture
+def netmiko_send_commands(netmiko_device, netmiko_mock_path):
+    def _mock(side_effects, existing_device=None, device=netmiko_device):
+        if existing_device is not None:
+            device = existing_device
+        with mock.patch.object(device, "_send_commands", autospec=True) as mock_send_commands:
+            mock_send_commands.side_effect = get_side_effects(netmiko_mock_path, side_effects)
+        device._send_commands = mock_send_commands
         return device
+
+    return _mock
+
+
+@pytest.fixture
+def netmiko_show(netmiko_device, netmiko_mock_path):
```

Comment:
Are you testing for single item (strings) vs multi-item?

Reply:
This only supports passing a single command currently. I started to update `get_side_effects` to support multiple commands, but then I decided that wasn't a current requirement, and that I should hold off on increasing the complexity until it was needed.